### ISSUE TITLE
LSP client failed with purescript and rust language server

### ISSUE THREAD

**Issue by DrJackilD** [neovim/neovim#16581]
### Neovim version (nvim -v)

0.6.0

### Language server name/version

 purs 0.14.5; rust-analyzer 

### Operating system/version

Mac OS 11.6

### :checkhealth

```markdown
lspconfig: 
========================================================================
  - ERROR: No healthcheck found for "lspconfig" plugin.

nvim: health#nvim#check
========================================================================
## Configuration
  - OK: no issues found

## Performance
  - OK: Build type: Release

## Remote Plugins
  - OK: Up to date

## terminal
  - INFO: key_backspace (kbs) terminfo entry: key_backspace=^H
  - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~
  - INFO: $TERM_PROGRAM='tmux'
  - INFO: $COLORTERM='truecolor'

## tmux
  - OK: escape-time: 10
  - INFO: Checking stuff
  - OK: focus-events: on
  - INFO: $TERM: screen-256color
```
```


### Steps to reproduce using "nvim -u minimal_init.lua"

Set up `name` to `purescriptls` and remove `cmd`, it's not needed for this server. Then open any `*.purs` file, for example with "Hello world":

```purescript
import Prelude
import Effect.Console (log)

greet :: String -> String
greet name = "Hello, " <> name <> "!"

main = log (greet "World")
```

Then set filetype to "purescript":
`:set ft=purescript`

After that you should see an error about client exit. No LSP client running (you could check `:LspInfo`)

### Expected behavior

Basically normal LSP working flow, like described in https://github.com/nwolverson/purescript-language-server

### Actual behavior

Client exit with error code 1

### Log file

https://gist.github.com/DrJackilD/b936bced18adcdaafd80afa9454bb732

**Comment by clason** [comment]
Issues with specific servers are off-topic for Neovim itself; ask on https://neovim.discourse.group or the Matrix chat room.
